packages/babel-helper-create-regexp-features-plugin/src/util.ts

```diff
@@ -71,6 +72,10 @@ export function canSkipRegexpu(
     return false;
   }
 
+  if (options.modifiers === "transform" && /\(\?.{1,10}:/.test(pattern)) {
```

Comment:
```suggestion
  if (options.modifiers === "transform" && /\(\?[\w-]+:/.test(pattern)) {
```
so that `(?:abc:)` doesn't match.